pagopa/io-dev-api-server
# docs: update README.md

## Short description
This PR updates the README.md of the repository.

## List of changes proposed in this pull request
- Restyles the content of the README.md;
- Removes outdated information and GIFs;
- Simplifies the Docker setup and includes nodenv instructions for the local setup;
- Removes the `yarn postinstall` command from the "Usage with the app" section which doesn't seem to be necessary anymore.

## How to test
You can check the markdown preview [here](https://github.com/pagopa/io-dev-api-server/tree/update-readme).
